### DISCUSSION TITLE
Beginner Questions

### DISCUSSION THREAD

**Discussion by mpinsley** [mermaid-js/mermaid#4441] [General]
Hi,

A few questions. Right now I am just playing with the Mermaid live.

1. How do we change colors etc., using the Live version. I have seen the CSS, but where do I put it, etc. 
2. Same question as above, but assume that I am in Notion or something like that; how do we add colors schemes, etc (a simple example would be great)
3. Is there a way to change the size of the output?
4. Can the output be printed on two sheets of paper? We have been using XLS to create Gantt charts. It is really a pain to deal with. I wanted to try and use Mermaid to solve the problem, but the output is so small, and I can not control where the page breaks are if I make it bigger. 

5. Slightly differently, is there a way to do swimlanes or something similar for business process mapping

Thanks

**Comment by sidharthv96** [comment]
1. The docs of many diagrams have styling section. eg. https://mermaid.js.org/config/theming.html, https://mermaid.js.org/syntax/flowchart.html#styling-and-classes, https://mermaid.js.org/syntax/classDiagram.html#styling
2. Depends on the diagram type. Example below.
```mermaid
flowchart LR
    id1(Start)-->id2(Stop)
    style id1 fill:#f9f,stroke:#333,stroke-width:4px
    style id2 fill:#bbf,stroke:#f66,stroke-width:2px,color:#fff,stroke-dasharray: 5 5
```
3. I believe there is a width/height option, but the SVG should be scalable manually too.
4. Nothing to handle printing is currently there, but something could be done based on your use case. 
5. Swimlanes might be coming soon (🤫)...

**Reply by mpinsley** [reply]
Hi

Thank you!  but in the live version. I saw the syntax, but I am not sure how to put the CSS in. Does it go directly into the code area? Does it go into the Config area?  And if it goes into Config, how do you deal with that when you are not using the Mermaid Live

two other questions
1. Is it possible to make the Sections bold?
2. Is it possible to make sure that the first Vertical line representing the first date only comes after the longest section name?

Thanks!

**Reply by sidharthv96** [reply]
1. In live editor, you can use the config section to set the colors. 
2. I'm not sure, sorry. 

This diagram has the style embedded in it.
```
flowchart LR
    id1(Start)-->id2(Stop)
    style id1 fill:#f9f,stroke:#333,stroke-width:4px
    style id2 fill:#bbf,stroke:#f66,stroke-width:2px,color:#fff,stroke-dasharray: 5 5
```

**Reply by mpinsley** [reply]
Thank you. So I have a pretty big Gantt Chart. here is a very small example 

I don't want to have to go through line by line to add the style
I want to be able to change the Title to be larger
I want to change the colors so active tasks are one color and done are another
And the background right now, which is blue and white and yellow. I want to change that

Is there a simple way of doing that? Or can you point me in the right direction?
What are the steps to using the configuration tool 

Sorry for so many questions. I have looked online but haven't found anything. i thought for sure there would be a video 

gantt
title  AUDIT PLAN 2023
dateFormat   MM/DD/YYYY
axisFormat   %m/%d
 
section Asst Deputy Controller   (JB)
Task 1                                                        : active    , 01/01/2023, 02/07/2023
Task 2                                                        : active    , 01/01/2023,   02/07/2023
Task 3                                                        : active    , 01/01/2023,   02/15/2023
Task 4                                                        : active    , 02/01/2023,   02/28/2023
Task 5                                                        : active    , 03/01/2023,   04/15/2023
Task 6                                                        : active    , 01/01/2023,   05/15/2023
Task 7                                                        : active    , 01/01/2023,   05/15/2023
Task 8                                                        : done      , 05/01/2023,   06/03/2023
Task 9                                                        : done      , 08/01/2023,   09/20/2023
Task 10                                                       : done      , 08/01/2023,   09/20/2023
Task 11                                                       : done      , 06/01/2023,   11/05/2023
Task 12                                                       : done      , 10/01/2023,   11/10/2023
Task 13                                                       : done      , 11/01/2023,   12/20/2023
Task 14                                                       : done      , 01/01/2023,   12/31/2023
Task 15                                                       : done      , 01/01/2023,   12/31/2023
 
section Auditor (AV)
Task 16                                                       : active    , 11/01/2022,   12/31/2022
Task 17                                                       : active    , 12/01/2022,   01/25/2023
Task 18                                                       : active    , 01/01/2023,   02/28/2023
Task 19                                                       : active    , 02/01/2023,   04/10/2023
Task 20                                                       : active    , 04/01/2023,   04/30/2023
Task 21                                                       : done      , 04/01/2023,   06/20/2023
Task 22                                                       : done      , 06/05/2023,   08/15/2023
Task 23                                                       : done      , 08/01/2023,   09/08/2023
Task 24                                                       : done      , 09/01/2023,   09/30/2023
Task 25                                                       : done      , 08/10/2023,   11/10/2023
Task 26                                                       : done      , 01/01/2023,   12/31/2023
Task 27                                                       : done      , 12/01/2023,   12/31/2023
Task 28                                                       : done      , 11/01/2023,   12/31/2023
Task 29                                                       : done      , 01/01/2023,   12/31/2023



</body>

</html>

**Reply by mpinsley** [reply]
ok so I changed teh heading to this. In live, I put it all in the Code section versus the "configure" section. I do not understand the config section yet. But I can not figure out how to add the font size for the Sections.

I also saw that when you download the image it only downloads what is seen in the preview screen. So if it isn't big enough, you can not zoom into it with another app


%%{init: {
--
'theme': 'forest',
'themeVariables': {
'primaryColor': '#BB2528',
'primaryTextColor': '#fff',
'primaryBorderColor': '#7C0000',
'lineColor': '#F8B229',
'secondaryColor': '#006100',
'tertiaryColor': '#fff'
}
}}%%
 
gantt
title  AUDIT PLAN 2023
dateFormat   MM/DD/YYYY
axisFormat   %m/%d



</body>

</html>

**Comment by SlyCooperking** [comment]
I'm using Obsidian for note taking, and I'm not sure how to properly scale the graphs I make. This is My markdown 

```mermaid
graph LR
subgraph Risk_Assessment
A[<ul><li>Perform risk categorization</li><li>Assess the level of impact</li><li>Determine the threat and risk levels</l></ul>]
end

subgraph Remediation
B[<ul><li>Prioritize remediation based on the risk ranking</li><li>Develop an action plan to implement the recommendation/remediation</li><li>Perform root cause analysis</li><li>Apply patches/fixes</li><li>CaptureLessons Learned</li><li>Conduct awareness training</li></ul>]
end

subgraph Verification 
C[<ul><li>Rescan of systems to identify if applied fix has remediated the vulnerability</li><li>Perfrom dynamic analysis</li><li>Review of attack surface</li></ul>]
end

subgraph Monitoring
D[<ul><li>Periodic vulnerability scan and assessment</li><li>Timely remediation of identified vulnerabilities</li><li>Intrusion detection and intrusion prevention logs</li><li>Implementation of policies, procedures, and controls</li></ul>]
end



Risk_Assessment --> Remediation --> Verification --> Monitoring
```
_
graph LR
subgraph Risk_Assessment
A[<ul><li>Perform risk categorization</li><li>Assess the level of impact</li><li>Determine the threat and risk levels</l></ul>]
end

subgraph Remediation
B[<ul><li>Prioritize remediation based on the risk ranking</li><li>Develop an action plan to implement the recommendation/remediation</li><li>Perform root cause analysis</li><li>Apply patches/fixes</li><li>CaptureLessons Learned</li><li>Conduct awareness training</li></ul>]
end

subgraph Verification 
C[<ul><li>Rescan of systems to identify if applied fix has remediated the vulnerability</li><li>Perfrom dynamic analysis</li><li>Review of attack surface</li></ul>]
end

subgraph Monitoring
D[<ul><li>Periodic vulnerability scan and assessment</li><li>Timely remediation of identified vulnerabilities</li><li>Intrusion detection and intrusion prevention logs</li><li>Implementation of policies, procedures, and controls</li></ul>]
end



Risk_Assessment --> Remediation --> Verification --> Monitoring_

It' doesn't want to fit within the confines of it's workspace. 
![image](https://github.com/mermaid-js/mermaid/assets/45924806/9ecf8787-4bc5-41ae-9280-c12f626e10d0)


Please let me know if there is a better way to do this, Also if there is a way to add spaces to subgraphs and why some subgraphs are interpreted as nodes. 

thanks, 
WC.

**Reply by sidharthv96** [reply]
This lies with Obsidian developers, as you can see, it's scaled properly in GitHub. 
![image](https://github.com/mermaid-js/mermaid/assets/10703445/1f9d2348-d625-479f-af25-61ed1e53d037)

**Reply by SlyCooperking** [reply]
Looking into it more it looks like Obsidian devs are waiting for 
https://github.com/mermaid-js/mermaid/issues/3462 to be solved. 

https://forum.obsidian.md/t/another-mermaid-dpi-scaling-bug/47699/7

Like the chicken and the egg problem?

**Comment by danyuandefine** [comment]
> I recently discovered an online drawing tool called PDDON that does not integrate mermaid. As a veteran markdown enthusiast, I particularly enjoy using mermaid. How can I suggest that authors of PDDON support mermaid syntax? 
>
> I would like to first generate a flowchart using mermaid syntax, and then import it into PDDON for formatting and beautification, as PDDON is indeed good at drawing related functions and I often use it, For example, the following figures were drawn using PDDON:

![drawings](https://pddon.github.io/pddon-win/images/ai.jpeg)
![drawworkspace](https://pddon.github.io/pddon-win/images/classDiagram.png)
